src/librustc_metadata/decoder.rs

```diff
@@ -52,8 +52,11 @@ pub struct DecodeContext<'a, 'tcx: 'a> {
     cdata: Option<&'a CrateMetadata>,
     from_id_range: IdRange,
     to_id_range: IdRange,
+
     // Cache the last used filemap for translating spans as an optimization.
     last_filemap_index: usize,
+
+    lazy_state: LazyState
```

Comment:
I feel like this lazy scheme needs a comment explaining how it works somewhere. Am I missing such a comment?